agent/api/container/status/managedagentstatus.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package status
+
+import (
+	"errors"
+	"strings"
+)
+
+const (
+	// ManagedAgentStatusNone is the zero state of a managed agent
+	ManagedAgentStatusNone ManagedAgentStatus = iota
+	// ManagedAgentCreated represents a managed agent which has dependencies in place
+	ManagedAgentCreated
+	// ManagedAgentRunning represents a managed agent that has started sucessfully
+	ManagedAgentRunning
+	// ManagedAgentStopped represents a managed agent that has stopped
+	ManagedAgentStopped
+)
+
+// ManagedAgentStatus is an enumeration of valid states in the managed agent lifecycle
+type ManagedAgentStatus int32
+
+var managedAgentStatusMap = map[string]ManagedAgentStatus{
```

Comment:
nit: to me makes more sense to reverse this map, then you can eliminate the for-loop from the String() function like:
```go
// String returns a human readable string representation of this object
func (mas ManagedAgentStatus) String() string {
    if s, ok := managedAgentStatusMap[mas]; ok {
        return s
    }
    return "NONE"
}
```